### DISCUSSION TITLE
companyName field value is coming null in BasketMgr.getCurrentBasket().getshipments().shippingaddress even though there is a companyName associated to it.

### DISCUSSION THREAD

**Discussion by himabindukadiyala** [SalesforceCommerceCloud/sfcc-ci#530] [Team Posts]
Hi, we are trying to display 'companyName' field corresponding to an address on Shipping page. But, the currentBasket retrieved using BasketMgr.getCurrentBasket(), is not populating returning any value for companyName field in BasketMgr.getCurrentBasket().getshipments().shippingaddress, I see this field coming as null. How do we populate this value? Is there any config to add this value such that BasketMgr.getCurrentBasket()getshipments().shippingaddress will return value for companyName. I can see comapnyName field value in req.currentCustomer.raw.address. Thank you!

**Comment by pravngaur** [comment]
Hi, all these fields are part of the standard address bean, once the
address is successfully saved we can validate the address in BM, look for
the particular customer and validate her address. I believe that it has to
do something with your custom logic, where you are persisting the user
address...

For a registered customer :
1) save an address and validate all the fields in BM.
2) IF all the fields are saved correctly then access them in storefront, if
any field is missing, check the logic where you are reading from system api
and setting the values in DTO....

Please debug your application logic, see where you are setting the variable
values...

**Comment by himabindukadiyala** [comment]
Thanks for the reply @pravngaur. Do you think we need to update this - https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/pull/271?

**Comment by pravngaur** [comment]
Yes it is very welll possible, check the model through which you are
getting the data back on page/response....
It might be possible that the companyName is not initialized in model..

Which isml are you working on?

**Comment by himabindukadiyala** [comment]
@pravngaur, I am getting the data from BasketMgr.getCurrentBasket().getshipments().shippingaddress. I am not able to see OOTB classes in my code base to debug how these are getting populated. In the shippingaddress object I am seeing companyName field, but the value is coming as null. I am using shippingAddress.isml to populate companyName, as this does not a value, it is not coming up. Also, to invoke isEquivalentAddress() method on an address from customer object, the address is not equivalent as the companyName in customer object has value but the companyName in shippinaddress object is null.

**Comment by pravngaur** [comment]
Are you using ootb sfra, i can do a quick debug..

**Comment by himabindukadiyala** [comment]
While checking this, I am not on OOTB SFRA. I verfied this in my local sandbox with out site. But, we didn't customize anything around BasketMgr.

**Comment by himabindukadiyala** [comment]
Can we see what is the code used for isEquivalentAddress() from OOTB classes? So that I can write my own function to exclude check on companyName and try to compare addresses from customerObject and BasketMgr.getCurrentBasket().getshipments().shippingaddress and get companyName from customerObject.

**Comment by pravngaur** [comment]
K. So, you are getting the companyName populated in the model object when
you dump the entire object on isml?

**Comment by himabindukadiyala** [comment]
no, we are not getting it in the model object.

**Comment by arvkumar4** [comment]
Hi Guys,

Could you please stop mailing everybody? Better take personal mail id do what you want to do.

Thanks,
Arvind

**Comment by GeneralPatten** [comment]
@arvkumar4 You can mute any discussion by clicking on the "mute the thread" link at the bottom of your emails.

**Comment by pravngaur** [comment]
If you are not getting it in model then you need to set it up explicitly,
override the model and include this additional attribute there.
